### PR TITLE
Integrate MuJoCo with Polymetis

### PR DESCRIPTION
**Pull request by laikhtewari**
# Description

This PR adds the MuJoCo manipulator which enables running robots in the MuJoCo simulator through Polymetis. 

Fixes # (issue)

[T122948315](https://www.internalfb.com/intern/tasks/?t=122948315)

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [x] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [x] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

If this PR introduces a change or fixes a bug, please add before and after screenshots (if this is causes a UX change) or before and after examples(this could be plain text, videos etc ) to demonstrate the change.

If this PR introduces a new feature, please also add examples or screenshots demonstrating the feature.

# Testing

Please describe the tests that you ran to verify your changes and how we can reproduce.

Follow the [Polymetis installation instructions](https://facebookresearch.github.io/fairo/polymetis/installation.html#from-source) or update the existing conda environment with `conda env update --file ./polymetis/environment.yml --prune`

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [ ] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have added relevant collaborators to review the PR before merge.
- [ ] [Polymetis only] I ran on hardware (1) all scripts in `tests/scripts`, (2) asv benchmarks.

### CODE REVIEW

**Comment by laikhtewari** [comment]
The current implementation requires specifying the robot model in both URDF and MJCF files in order to use the MuJoCo manipulator. [This issue](https://github.com/facebookresearch/fairo/issues/1332) discusses using a single file to specify the robot model

File: polymetis/polymetis/python/polysim/envs/mujoco_manipulator.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    def apply_joint_torques(self, torques: np.ndarray):
+        """
+        input:
+            np.ndarray: Desired torques
+        Returns:
+            np.ndarray: final applied torque
+        """
+        self.prev_torques_commanded = torques
+
+        applied_torques = np.clip(torques, -self.torque_limits, self.torque_limits)
```
Comment (1heart) [inline polymetis/polymetis/python/polysim/envs/mujoco_manipulator.py:148]:
```suggestion
        applied_torques = np.clip(torques, -self.torque_limits, self.torque_limits)
        self.prev_torques_applied = applied_torques
```

File: polymetis/polymetis/python/polysim/envs/mujoco_manipulator.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+        self.robot_data.ctrl = applied_torques
+        mujoco.mj_step(self.robot_model, self.robot_data)
+
+        if self.gui:
+            self.render()
+
+        return applied_torques
+
+    def compute_inverse_dynamics(
```
Comment (1heart) [inline polymetis/polymetis/python/polysim/envs/mujoco_manipulator.py:168]: Quite important to unit-test this method (though you should test the others as well) by ensuring result is comparable (within epsilon) against e.g. Pinocchio or PyBullet output.

File: polymetis/polymetis/python/polysim/envs/mujoco_manipulator.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        self.robot_model_cfg = robot_model_cfg
+        self.robot_description_path = get_full_path_to_urdf(
+            self.robot_model_cfg.robot_description_path
+        )
+        robot_desc_mjcf_path = (
+            os.path.splitext(self.robot_description_path)[0] + ".mjcf"
+        )
+        assert os.path.exists(
+            robot_desc_mjcf_path
+        ), f"No MJCF file found. Create an MJCF file at {robot_desc_mjcf_path} to use the MuJoCo simulator."
```
Comment (1heart) [inline polymetis/polymetis/python/polysim/envs/mujoco_manipulator.py:38]: This is a reasonable solution for now -- just make sure to document the need for both urdf and mjcf. The logic is not immediately apparent that you need both.

**Review by 1heart — COMMENTED** [review]
Generally looks like the right direction! Please make sure to add unit tests, especially covering the numerical methods. (You can find an example comparing Bullet to Pinocchio [here](https://github.com/facebookresearch/fairo/blob/main/polymetis/polymetis/tests/python/polymetis/test_torchscript_pinocchio.py#L143))

File: polymetis/polymetis/tests/python/polysim/test_env.py [hunk 4]
```diff
@@ -55,6 +56,46 @@
     },
 )
 
+franka_panda_abs = OmegaConf.create(
+    {
+        "robot_description_path": "/root/project/polymetis/polymetis/data/franka_panda/panda_arm.urdf",
```
Comment (1heart) [inline polymetis/polymetis/tests/python/polysim/test_env.py:61]: Yeah... let's not hardcode the directory structure of a random circleci image into our unit tests. This is pretty much guaranteed to fail locally.

**Review by 1heart — APPROVED** [review]
lgtm!

**Pull request merged**